File: Core/src/main/java/jokatu/ui/DialogFormBuilder.java
package jokatu.ui;

import ophelia.builder.Builder;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

public class DialogFormBuilder implements Builder<Form, DialogFormBuilder> {

	private final List<FormElement> elements = new ArrayList<>();

	@NotNull
	@Override
	public Form build() {
		return new Form(elements);
	}

	@NotNull
	public DialogFormBuilder withDiv(@NotNull String label) {
		elements.add(new FormDiv(label));
		return this;
	}

	@NotNull
	public DialogFormBuilder withField(
			@NotNull FormField field
	) {
		elements.add(field);
		return this;
	}

	@NotNull
	public DialogFormBuilder withFields(@NotNull List<FormField> fields) {
		this.elements.addAll(fields);
		return this;
	}

	@NotNull
	@Override
	public DialogFormBuilder noöp() {
		return this;
	}
}

File: Core/src/main/kotlin/jokatu/ui/DialogFormBuilder.kt
package jokatu.ui

import ophelia.builder.Builder
import java.util.*

class DialogFormBuilder : Builder<Form, DialogFormBuilder> {

	private val elements = ArrayList<FormElement>()

	override fun build(): Form {
		return Form(elements)
	}

	fun withDiv(label: String): DialogFormBuilder {
		elements.add(FormDiv(label))
		return this
	}

	fun withField(
			field: FormField
	): DialogFormBuilder {
		elements.add(field)
		return this
	}

	fun withFields(fields: List<FormField>): DialogFormBuilder {
		this.elements.addAll(fields)
		return this
	}

	override fun noöp(): DialogFormBuilder {
		return this
	}
}
